### ISSUE TITLE
NaN id loss while training your custom dataset

### ISSUE THREAD

**Issue by yangkaizhang** [ifzhang/FairMOT#546]
just cheek your labels files in "./labels_with_ids".Some custom datasets list their anns in sequeence of frame_id instead of track_id ,which is inappropriate for origin "gen_labels_[ ].py(s)" in this program(especially those modified MOT format) and result in incorrect id_indexs after convert(normally too large).

**Comment by ZWL706** [comment]
你好，我想请问下你是否有使用过"ETHZ"数据集进行训练，我发现虽然也出现了id_loss=nan，但是原因似乎并不是跟你的一样，而是由于作者使用交叉熵损失`self.IDLoss = nn.CrossEntropyLoss(ignore_index=-1)`这个设置意味着所有标签为-1的条目在计算损失时都会被忽略。

**Comment by yangkaizhang** [comment]
> 你好，我想请问下你是否有使用过"ETHZ"数据集进行训练，我发现虽然也出现了id_loss=nan，但是原因似乎并不是跟你的一样，而是由于作者使用交叉熵损失`self.IDLoss = nn.CrossEntropyLoss(ignore_index=-1)`这个设置意味着所有标签为-1的条目在计算损失时都会被忽略。

对ReID数据集，在训练时可以不加入id损失；我的repo只适用错误生成的不匹配的标签导致id loss在特定epoch中发生梯度爆炸

**Comment by BansheeZZ** [comment]
我在训练自制水下目标数据集时也容易发生id_loss为nan，不太懂为什么
